src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -246,13 +246,45 @@ HeapWord* ShenandoahFreeSet::allocate_single(ShenandoahAllocRequest& req, bool&
   return nullptr;
 }
 
+size_t get_usable_free_words(size_t free) {
```

Comment:
Yes, I agree. I would like something like a strong typedef here to prevent mixing bytes and words in the same expression. If you look closely at the [original code](https://github.com/openjdk/shenandoah/pull/260/files#diff-46de849c458b5bfca09d1c8134e9d71d131faac923b39d44db2fe27d168941aeL370), the path for in-elastic TLABs mixed up words and bytes.

Reply:
I wrote it this way to reduce duplication in the callers.